network-area-diagram/src/main/java/com/powsybl/nad/Param.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.nad;
+
+import com.powsybl.iidm.network.Network;
+import com.powsybl.nad.build.iidm.IdProvider;
+import com.powsybl.nad.layout.LayoutFactory;
+import com.powsybl.nad.layout.LayoutParameters;
+import com.powsybl.nad.svg.LabelProvider;
+import com.powsybl.nad.svg.StyleProvider;
+import com.powsybl.nad.svg.SvgParameters;
+
+/**
+ *
+ * @author Sophie Frasnedo <sophie.frasnedo at rte-france.com>
+ */
+public class Param {
+
+    SvgParameters svgParameters;
+    LayoutParameters layoutParameters;
+    ParamBuilder.StyleProviderFactory styleProviderFactory;
+    ParamBuilder.LabelProviderFactory labelProviderFactory;
+    LayoutFactory layoutFactory;
+    ParamBuilder.IdProviderFactory idProviderFactory;
+
+    public Param(SvgParameters svgParameters, LayoutParameters layoutParameters, ParamBuilder.StyleProviderFactory styleProviderFactory, ParamBuilder.LabelProviderFactory labelProviderFactory, LayoutFactory layoutFactory, ParamBuilder.IdProviderFactory idProviderFactory) {
+        this.svgParameters = svgParameters;
+        this.layoutParameters = layoutParameters;
+        this.styleProviderFactory = styleProviderFactory;
+        this.labelProviderFactory = labelProviderFactory;
+        this.layoutFactory = layoutFactory;
+        this.idProviderFactory = idProviderFactory;
+    }
+
+    public SvgParameters getSvgParameters() {
+        return svgParameters;
+    }
+
+    public LayoutParameters getLayoutParameters() {
+        return layoutParameters;
+    }
+
+    public StyleProvider createStyleProvider(Network network) {
```

Comment:
Following the discussion about labelProviderFactory, which makes the getter a bit difficult to handle, I think that, to have this logic applied to the labelProvider, we should have a simplified interface, for sld for instance:
```java
    @FunctionalInterface
    public interface LabelProviderFactory {
        LabelProvider create(Network network);
    }
```

And then the default value would be defined in the builder as:
```java
LabelProviderFactory labelProviderFactory = n -> new DefaultLabelProvider(n, componentLibrary, layoutParameters, svgParameters);
```

Reply:
Ok, what I suggested isn't really a good idea, because when you define the label provider with the builder, it becomes something very clumsy:
```java
param = new ParamBuilder()
    .withLabelProviderFactory(n -> 
        new DefaultLabelProvider(n, componentLibrary, layoutParameters, svgParameters))
    .build();
```
Which forces you to define the componentLibrary, the layoutParameters and the svgParameters even if you want the default one.

Reply:
Agreed. To me, there is no obvious solution to get rid of this createLabelProvider in the SldParameters and NadParameters classes.